src/AppBundle/Model/Event.php

```diff
@@ -365,17 +372,67 @@ public function setTimezone($timezone)
      **************/
 
     /**
-     * Are there any categories set on EventWikis associated with this Event?
-     * @return bool
+     * Get categories belonging to this Event.
+     * @return ArrayCollection of EventCategories.
      */
-    public function hasCategories()
+    public function getCategories()
     {
-        foreach ($this->wikis as $wiki) {
-            if (count($wiki->getCategories()) > 0) {
-                return true;
-            }
+        return $this->categories;
+    }
+
+    /**
+     * Get the number of categories belonging to this Event.
+     * @return int
+     */
+    public function getNumCategories()
+    {
+        return count($this->categories);
+    }
+
+    /**
+     * Get the titles of categories belonging to this Event that are for the specified wiki.
+     * @param EventWiki $wiki
+     * @return string[]
+     */
+    public function getCategoryTitlesForWiki(EventWiki $wiki)
+    {
+        return $this->categories->filter(function (EventCategory $category) use ($wiki) {
+            return $category->getDomain() === $wiki->getDomain();
+        })->map(function (EventCategory $category) {
+            return $category->getTitle();
+        })->toArray();
```

Comment:
My PHP is admittedly rusty but this code is very confusing to me. What is it doing?

Reply:
For starters, this is using Doctrine's [ArrayCollection](https://www.doctrine-project.org/api/collections/latest/Doctrine/Common/Collections/ArrayCollection.html) to traverse the arrays, effectively the object-oriented style of `array_filter` and `array_map`. Not sure if that was the confusing part. On a high level, this method first gets categories on the event that are configured for the given wiki, then loops through again to get their titles. I'll add some comments for clarity.